PublicFolders/ValidateEXOPFDumpster.ps1

```diff
@@ -0,0 +1,439 @@
+        WritetoScreenANDlog -Issue $Issue -Fix $Fix
+    }
+}
+function ValidateMEPFGuid {
+    param([PSCustomObject]$PublicFolderInfo)
+    #validate if MailRecipientGuid parameter is found empty/null
+    if ($PublicFolderInfo.Publicfolder.MailEnabled -eq $true) {
+        $Hasvalue=$PublicFolderInfo.Publicfolder.MailRecipientGuid.Guid
+        if ($null -eq $Hasvalue -or $Hasvalue -eq "00000000-0000-0000-0000-000000000000") {
+            $Fix= "FIX --> Please follow the following article https://aka.ms/EnableMPF to mail enable the affected public folder to generate a GUID over MailRecipientGuid parameter, validate MailRecipientGuid parameter has a GUID using https://aka.ms/getpf article then mail disable the affected public folder back again using https://aka.ms/disablempf article to be able to remove the public folder as requested."
+            $Issue="Mail-enabled public folder $($PublicFolderInfo.Publicfolder.Identity) is unhealthy e.g MailRecipientGuid parameter is found empty/null which is a blocker for deletion operations over the public folder"
+            WritetoScreenANDlogDlog -Issue $Issue -Fix $Fix
+        }
+    }
+}
+function ExtractLog {
+    param([PSCustomObject]$PublicFolderInfo)
+    #create zip file for logs folder
+    if (!(Test-Path  "$ExportPath\logs_$ts")) {
+        mkdir "$ExportPath\logs_$ts" -Force | Out-Null
+    }
+    $PublicFolderInfo | Export-Clixml -Path "$ExportPath\logs_$ts\PublicFolderInfo.xml"
+}
+#Intro
+$ts = Get-Date -Format yyyyMMdd_HHmmss
+Write-Host $ExportPath
+if ($null -eq $ExportPath -or $ExportPath -eq "") {
+    $ExportPath = "$env:USERPROFILE\Desktop\ValidatePFDumpster\ValidatePFDumpster_$ts"
+    mkdir $ExportPath -Force | Out-Null
+} else {
+    if (!(Test-Path -Path $ExportPath)) {
+        Write-Host "The specified folder location $ExportPath is not existing, please re-run the script and ensure to enter a valid path or leave the ExportPath unassigned" -ForegroundColor Red
+        exit
+    } else {
+        $ExportPath = "$ExportPath\ValidatePFDumpster\ValidatePFDumpster_$ts"
+        mkdir $ExportPath -Force | Out-Null
+    }
+}
+
+
+[string]$Description = "This script illustrates issues related to deleting public folder items or removing the public folder on Publicfolder $Pfolder, BLOCKERS will be reported down, please ensure to mitigate them!`n"
+Write-Host $Description -ForegroundColor Cyan
+$Description | Out-File $ExportPath\$Script:ReportName -Append
+
+#Connect to EXO PS
+$Sessioncheck = Get-PSSession | Where-Object { $_.Name -like "*Exchangeonline*" -and $_.State -match "opened" }
+if ($null -eq $Sessioncheck) {
+    Connect2EXO
+}
+
+#Main Function
+$PublicFolderInfo=GetPublicFolderInfo($Pfolder)
+#if the issue is related to a user who is not able to delete an item inside a public folder
+if ($Affecteduser -notlike "" -or $Affecteduser -notlike $null) {
```

Comment:
You can use `[string]::IsNullOrEmpty($Affecteduser)` here instead

Reply:
Thank you for the advice I modified that statement, but I messed up something I was trying to squash all my commits to 1 commit as you showed me earlier, but I ended up adding 2 more commits!
I replaced the word "Pick" with letter "s" then saved the file then from GitHub desktop I did Pull from origin then Push, any idea what did I do wrong?

C:\Users\haembab\Documents\GitHub\CSS-Exchange>git rebase -i Head~5
[detached HEAD fd7422ec] Add new script that troubleshoot public folder deletion failures
 Date: Tue Nov 8 16:36:52 2022 +0200
 10 files changed, 518 insertions(+)
 create mode 100644 PublicFolders/ValidateEXOPFDumpster.ps1
 create mode 100644 docs/PublicFolders/PFDumps.jpg
 create mode 100644 docs/PublicFolders/PFDumps2.jpg
 create mode 100644 docs/PublicFolders/PFDumps3.jpg
 create mode 100644 docs/PublicFolders/PFDumps4.jpg
 create mode 100644 docs/PublicFolders/PFDumps5.jpg
 create mode 100644 docs/PublicFolders/PFDumps6.jpg
 create mode 100644 docs/PublicFolders/PFDumpswithoutparam.jpg
 create mode 100644 docs/PublicFolders/ValidateEXOPFDumpster.md
Successfully rebased and updated refs/heads/haembab-validatepfdumspterscript.